### PR TITLE
Adds in Bola, Other Recipe Tweaks

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports @PKPenguin321 bola stuff: https://github.com/tgstation/tgstation/pull/15780

Bola are basically throwable legcuffs that trigger when they hit the person, but are super easy to resist out of/get off. These replace the horrendous implementation of bolas that we currently have.

You can make them by attaching 6 metal sheets to cablecuffs or personal crafting.

Sec also start with some e-bola (kek) in the sec vend; sec belts can hold/carry them

Other
- Halves the time of most tablecrafting recipes, in line with TG 
- Makes a few adjustments to improvised shells; they'll need more/rarer reagents, but should behave a bit more consistently
- Increases fire stack amount on incendiary shells
- Can make spears with personal crafting now

Fixes
- Fixes improved laser slugs not actually firing a laser
- Fixes a few crafting recipes not having their category set

:cl: Fox McCloud
fix: Fixes a few crafting recipes not having the proper category
fix: Fixes laser slugs not firing lasers
tweak: Adjusts the damage/consistency of improvised slugs
add: Adds in craftable bolas; throw them at someone to slow them down
add: Secvend starts off with e-bolas
add: Can make spears with personal crafting now
/:cl:

### CODE REVIEW

**Comment by Lady-Luck** [comment]
Oh man I was just talking about bolas in Discord yesterday

**Comment by IcyV** [comment]
>Sec also start with some e-bola

Emagged versions should totally cut off someone's feet.

In seriousness, though, this sounds like it will make chases a lot more interesting. A game of bolas versus tasers.

File: code/game/objects/items/weapons/legcuffs.dm [hunk 1]
```diff
@@ -122,105 +127,85 @@
...
+	var/weaken = 0
+
+/obj/item/weapon/restraints/legcuffs/bola/throw_impact(atom/hit_atom)
+	if(..() || !iscarbon(hit_atom))//if it gets caught or the target can't be cuffed,
+		return//abort
+	var/mob/living/carbon/C = hit_atom
+	if(!C.legcuffed)
+		visible_message("<span class='danger'>\The [src] ensnares [C]!</span>")
+		C.legcuffed = src
+		src.loc = C
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/legcuffs.dm]: forceMove

**Comment by FreeStylaLT** [comment]
img

File: code/game/objects/items/weapons/legcuffs.dm [hunk 2]
```diff
@@ -122,105 +127,85 @@
...
+	gender = NEUTER
+	origin_tech = "engineering=3;combat=1"
+	var/weaken = 0
+
+/obj/item/weapon/restraints/legcuffs/bola/throw_impact(atom/hit_atom)
+	if(..() || !iscarbon(hit_atom))//if it gets caught or the target can't be cuffed,
+		return//abort
+	var/mob/living/carbon/C = hit_atom
+	if(!C.legcuffed)
+		visible_message("<span class='danger'>\The [src] ensnares [C]!</span>")
```
Comment (TheDZD) [inline code/game/objects/items/weapons/legcuffs.dm]: Extraneous `\The`

File: code/game/objects/items/weapons/legcuffs.dm [hunk 3]
```diff
@@ -122,105 +127,85 @@
...
+		C.legcuffed = src
+		src.loc = C
+		C.update_inv_legcuffed()
+		feedback_add_details("handcuffs","B")
+		to_chat(C, "<span class='userdanger'>\The [src] ensnares you!</span>")
```
Comment (TheDZD) [inline code/game/objects/items/weapons/legcuffs.dm]: Another extraneous `\the`.

**Comment by Earthdivine** [comment]
You triggered me, @FreeStylaLT .

**Comment by PKPenguin321** [comment]
:+1:

**Comment by Spacemanspark** [comment]
Been wanting the ability to make bolas for awhile.
:+1:

**Comment by Da-Dman234** [comment]
Just out of curiosity, does this fix the Bola Launcher that Robotics can build as well?

**Comment by Fox-McCloud** [comment]
It works, yes.

File: code/game/objects/items/weapons/legcuffs.dm [hunk 4]
```diff
@@ -122,105 +127,85 @@
...
+	gender = NEUTER
+	origin_tech = "engineering=3;combat=1"
+	var/weaken = 0
+
+/obj/item/weapon/restraints/legcuffs/bola/throw_impact(atom/hit_atom)
+	if(..() || !iscarbon(hit_atom))//if it gets caught or the target can't be cuffed,
+		return//abort
+	var/mob/living/carbon/C = hit_atom
+	if(!C.legcuffed)
+		visible_message("<span class='danger'>The [src] ensnares [C]!</span>")
```
Comment (tigercat2000) [inline code/game/objects/items/weapons/legcuffs.dm]: uhh i think DZD meant remove `the` entirely
```
visible_message("<span class='danger'>[src] ensnares [C]!</span>")
```
same for below

Comment (TheDZD) [inline code/game/objects/items/weapons/legcuffs.dm]: Tiger is correct.

**Comment by Allfd** [comment]
I am... Had no idea we had these.

**Comment by taukausanake** [comment]
Overload Improvised Shells need plasma dust now? I mean I guess, okay but seems a bit overkill. Then again I don't know how strong those shells are

**Pull request merged**